src/config.ts

```diff
@@ -10,12 +10,25 @@ export enum TokenType {
   native = 'native',
 }
 
-export type SyntheticConfig = {
-  type: TokenType.synthetic | TokenType.syntheticUri;
+export type TokenMetadata = {
   name: string;
   symbol: string;
   totalSupply: ethers.BigNumberish;
 };
+
+export type ERC20Metadata = TokenMetadata & {
+  decimals: number;
+};
+
+export const isTokenMetadata = (metadata: any): metadata is TokenMetadata =>
+  metadata.name && metadata.symbol && metadata.totalSupply !== undefined; // totalSupply can be 0
```

Comment:
differentiating via some type prop instead would be nice but def don't block on that